pkg/ipamd/datastore/data_store.go

```diff
@@ -1198,24 +1122,15 @@ func (ds *DataStore) UnassignPodIPAddress(ipamKey IPAMKey) (e *ENI, ip string, d
 
 	eni, availableCidr, addr := ds.eniPool.FindAddressForSandbox(ipamKey)
 	if addr == nil {
-		// This `if` block should be removed when the CRI
-		// migration code is finally removed.  Leaving a
-		// compile dependency here to make that obvious :P
-		var _ = checkpointMigrationPhase
-
-		// If the entry was discovered by querying CRI at
-		// restart rather than by observing an ADD operation
-		// directly, then we won't have captured the true
-		// networkname/ifname.
+		// If the entry is not present in state file, check if it is present under placeholder value.
+		// This scenario could happen if the pod was created by an older CNI version back when CRI read was done.
 		ds.log.Debugf("UnassignPodIPAddress: Failed to find IPAM entry under full key, trying CRI-migrated version")
 		ipamKey.NetworkName = backfillNetworkName
 		ipamKey.IfName = backfillNetworkIface
 		eni, availableCidr, addr = ds.eniPool.FindAddressForSandbox(ipamKey)
-	}
```

Comment:
why is this if removed..

Reply:
The if condition below had the exact same check, so we don't need two if blocks, just one

Reply:
Ah sorry, fixed now